### ISSUE TITLE
Replace vague links to actual names of the pages/sections in docs

### ISSUE THREAD

**Issue by YarShev** [modin-project/modin#4085]
We should replace vague links to actual names of the pages/sections in docs to be clearer.

An example would be in https://modin.readthedocs.io/en/stable/getting_started/examples.html where we say the following.
```
To learn more about advanced usage for Modin, please refer to :doc:`this section </usage_guide/index>`.
```
This should be replaced as follows.
```
To learn more about advanced usage for Modin, please refer to :doc:`Usage Guide </usage_guide/index>` section.
```

**Comment by ahmadjubair33** [comment]
Hi can I modify this?

**Comment by anmyachev** [comment]
Closed by #4096
